### PR TITLE
Avoid building ONNX of every history ONNX versions in our CI

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 
1. Avoid building ONNX of every history ONNX versions in our CI, it is costly and easy to fail.
2. Run docker command without sudo. Previously the user is not in docker group,  now Azure DevOps Service have added it in.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
     - task: CmdLine@2
       inputs:
         script: |
-          sudo --preserve-env docker run --rm --volume $(Build.SourcesDirectory):/onnxruntime_src --volume $(Build.BinariesDirectory):/build --volume /data/models:/build/models:ro -e NIGHTLY_BUILD onnxruntime-centos6 /usr/bin/python3.6 /onnxruntime_src/tools/ci_build/build.py --build_dir /build --config Release --skip_submodule_sync  --parallel --build_shared_lib --use_openmp --cmake_path /usr/bin/cmake --ctest_path /usr/bin/ctest
+          sudo --preserve-env docker run --rm --volume /data/onnx:/data/onnx --volume $(Build.SourcesDirectory):/onnxruntime_src --volume $(Build.BinariesDirectory):/build --volume /data/models:/build/models:ro -e NIGHTLY_BUILD onnxruntime-centos6 /usr/bin/python3.6 /onnxruntime_src/tools/ci_build/build.py --build_dir /build --config Release --skip_submodule_sync  --parallel --build_shared_lib --use_openmp --cmake_path /usr/bin/cmake --ctest_path /usr/bin/ctest
```
Comment (thiagocrepaldi) [inline tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml:18]: How about making /data/onnx read-only to make sure the containers will not corrupt it between runs?

Something like `--volume /data/onnx:/data/onn:ro`

File: tools/ci_build/github/azure-pipelines/java-api-packaging-pipelines-gpu.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ jobs:
...
       inputs:
         script: |
-          sudo docker run --gpus all -e NVIDIA_VISIBLE_DEVICES=all --rm --volume $(Build.SourcesDirectory):/onnxruntime_src --volume $(Build.BinariesDirectory):/build --volume /data/models:/build/models:ro -e NIGHTLY_BUILD onnxruntime-centos6-gpu /usr/bin/python3.6 /onnxruntime_src/tools/ci_build/build.py --build_dir /build --config Release --skip_submodule_sync  --parallel --build_java --build_shared_lib --cmake_path /usr/bin/cmake --ctest_path /usr/bin/ctest --use_cuda --cuda_version=10.1 --cuda_home=/usr/local/cuda-10.1  --cudnn_home=/usr/local/cuda-10.1
+          sudo docker run --gpus all -e NVIDIA_VISIBLE_DEVICES=all --rm --volume /data/onnx:/data/onnx:ro --volume $(Build.SourcesDirectory):/onnxruntime_src --volume $(Build.BinariesDirectory):/build --volume /data/models:/build/models:ro -e NIGHTLY_BUILD onnxruntime-centos6-gpu /usr/bin/python3.6 /onnxruntime_src/tools/ci_build/build.py --build_dir /build --config Release --skip_submodule_sync  --parallel --build_java --build_shared_lib --cmake_path /usr/bin/cmake --ctest_path /usr/bin/ctest --use_cuda --cuda_version=10.1 --cuda_home=/usr/local/cuda-10.1  --cudnn_home=/usr/local/cuda-10.1
```
Comment (edgchen1) [inline tools/ci_build/github/azure-pipelines/java-api-packaging-pipelines-gpu.yml:17]: noticed a few sudo docker run's with -e X (e.g., -e NIGHTLY_BUILD). probably need sudo --preserve-env for that to work properly

File: tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml [hunk 3]
```diff
@@ -15,7 +15,7 @@ jobs:
     - task: CmdLine@2
       inputs:
         script: |
-          sudo --preserve-env docker run --rm --volume $(Build.SourcesDirectory):/onnxruntime_src --volume $(Build.BinariesDirectory):/build --volume /data/models:/build/models:ro -e NIGHTLY_BUILD onnxruntime-centos6 /usr/bin/python3.6 /onnxruntime_src/tools/ci_build/build.py --build_dir /build --config Release --skip_submodule_sync  --parallel --build_shared_lib --use_openmp --cmake_path /usr/bin/cmake --ctest_path /usr/bin/ctest
+          sudo --preserve-env docker run --rm --volume /data/onnx:/data/onnx:ro --volume $(Build.SourcesDirectory):/onnxruntime_src --volume $(Build.BinariesDirectory):/build --volume /data/models:/build/models:ro -e NIGHTLY_BUILD onnxruntime-centos6 /usr/bin/python3.6 /onnxruntime_src/tools/ci_build/build.py --build_dir /build --config Release --skip_submodule_sync  --parallel --build_shared_lib --use_openmp --cmake_path /usr/bin/cmake --ctest_path /usr/bin/ctest
```
Comment (edgchen1) [inline tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml:18]: nit: can we have some maximum line length for these yml files too?

File: tools/ci_build/github/azure-pipelines/java-api-packaging-pipelines-gpu.yml (hunk 2 above)
Comment (snnn) [reply tools/ci_build/github/azure-pipelines/java-api-packaging-pipelines-gpu.yml]: Thank you for pointing it out.
I changed it to run docker command without sudo. Previously the user is not in docker group, now Azure DevOps Service have added it in.

File: tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml (hunk 3 above)
Comment (edgchen1) [reply tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml]: thanks for updating these lines. this is a general comment, but it doesn't have to be completely addressed in this PR.

**Pull request merged**